registration/include/pcl/registration/gicp.h

```diff
@@ -423,21 +430,39 @@ class GeneralizedIterativeClosestPoint
   void
   applyState(Eigen::Matrix4f& t, const Vector6d& x) const;
 
-  /// \brief optimization functor structure
-  struct OptimizationFunctorWithIndices : public BFGSDummyFunctor<double, 6> {
-    OptimizationFunctorWithIndices(const GeneralizedIterativeClosestPoint* gicp)
-    : BFGSDummyFunctor<double, 6>(), gicp_(gicp)
+  /// \brief optimization functor structure for translation part
+  struct OptimizationFunctorWithIndicesTranslation
+  : public BFGSDummyFunctor<double, 3> {
+    OptimizationFunctorWithIndicesTranslation(GeneralizedIterativeClosestPoint* gicp)
```

Comment:
Need to update `gicp_->current_transformation_` so `const` from `const GeneralizedIterativeClosestPoint*` removed.